### PR TITLE
Improve Istio role

### PR DESCRIPTION
**Pull request by geoand**
This shouldn't be merged yet, since it's a work in progress

### CODE REVIEW

**Comment by geoand** [comment]
@cmoulliard This should ready for review

File: ansible/extra_vars.yml [hunk 1]
```diff
@@ -18,37 +18,4 @@ target_platform: local
...
-
-# The GitHub token to use when calling GitHub's API
-# The token is needed in order to avoid getting Rate Limited by GitHub's API
-# Rate Limiting can easily occur in large corporate environments
-# that make outgoing calls through a small range of IPs
-# If the token is not set, then no API Token will be used
-github_api_token: ""
-
-istio:
-  release_tag_name: "0.6.0"
```
Comment (cmoulliard) [inline ansible/extra_vars.yml]: This var should be then added to the default value of the role ?

Comment (geoand) [reply ansible/extra_vars.yml]: The `github_api_token` is not needed at all. The `release_tag_name` was added in `ansible/playbook/roles/install_istio/vars/main.yml`

Comment (cmoulliard) [reply ansible/extra_vars.yml]: But I override the value within the `extra_vars.yml` file. So keep it please like the dest folder which is different.

Comment (geoand) [reply ansible/extra_vars.yml]: I have no problem changing things back to the way they were, but I'm trying to understand why we would need to.
I might be missing something, but why would we need it in `extra_vars` when it's already been defined in `vars/main.yml` that I have added? The benefit of having it in `vars/main.yml` is that the role works even without having to specify the `extra_vars` file.
Furthermore, why does the `dest` folder need to be set specifically to `/home/istio`?

Comment (cmoulliard) [reply ansible/extra_vars.yml]: > dest folder

it is used to install `istio` distro containing `istioctl` client, examples, ...

Comment (geoand) [reply ansible/extra_vars.yml]: Yes I know, my question was basically whether it matters were it's installed?

I'll go ahead and change it back :)

Comment (geoand) [reply ansible/extra_vars.yml]: Changed in `vars/main.yml`

**Comment by geoand** [comment]
@cmoulliard Are we going to take this? If not, what changes should be made?

**Comment by geoand** [comment]
I tested this both for Openshift `3.7.2` and `3.9.0.alpha3` using: 
`ansible-playbook -i inventory/cloud_host playbook/post_installation.yml -e "@extra_vars.yml" --tags istio` 
and it works as expected

**Comment by geoand** [comment]
I will need to rebase this on master before we can merge

**Comment by geoand** [comment]
Ready to be merged :)

**Comment by geoand** [comment]
This now contains commits to get Istio working with the mutating webhook.
We should go ahead and merge it @cmoulliard

**Pull request merged**